storage/layers.go

```diff
@@ -467,6 +481,8 @@ func (r *layerStore) Unmount(id string) error {
 	}
 	if layer, ok := r.byname[id]; ok {
 		id = layer.ID
+	} else if longid, err := r.idindex.Get(id); err == nil {
```

Comment:
can you have a function which fisrt check the `r.byname[id]` and the the truncindex to not repeat this else if over? skip this if you think it's not worth.